apps/prairielearn/src/tests/questionSharing.test.ts

```diff
@@ -68,10 +68,31 @@ async function accessSharedQuestionAssessment(course_id: string) {
 const baseDir = tmp.dirSync().name;
 const sharingCourseOriginDir = path.join(baseDir, 'courseOrigin');
 const sharingCourseLiveDir = path.join(baseDir, 'courseLive');
-const gitOptions = {
+const gitOptionsOrigin = {
   cwd: sharingCourseOriginDir,
   env: process.env,
 };
+const gitOptionsLive = {
+  cwd: sharingCourseLiveDir,
+  env: process.env,
+};
+async function commitAndPullSharingCourse() {
+  await execa('git', ['add', '-A'], gitOptionsOrigin);
+  await execa('git', ['commit', '-m', 'Add sharing set'], gitOptionsOrigin);
+  await execa('git', ['pull'], gitOptionsLive);
+  const syncResults = await syncUtil.syncCourseData(sharingCourseLiveDir);
+  assert(syncResults.status === 'complete' && !syncResults.hadJsonErrorsOrWarnings);
```

Comment:
Nit: typically it's better to use `assert.equal` and other ones that take an expected and actual value, as Mocha will rovide more informative error messages.

For example:

```js
import { assert } from 'chai';

describe('example', () => {
  it('uses assert', () => {
    assert('foo' === 'bar');
  });

  it('uses assert.equal', () => {
    assert.equal('foo', 'bar');
  });
});
```

```
  1) example
       uses assert:
     
  AssertionError
      at Context.<anonymous> (src/tests/example.test.ts:5:5)
      at process.processImmediate (node:internal/timers:483:21)
      at process.callbackTrampoline (node:internal/async_hooks:130:17)

  2) example
       uses assert.equal:

      AssertionError: expected 'foo' to equal 'bar'
      + expected - actual

      -foo
      +bar
      
      at Context.<anonymous> (src/tests/example.test.ts:9:12)
      at process.processImmediate (node:internal/timers:483:21)
      at process.callbackTrampoline (node:internal/async_hooks:130:17)
```

Reply:
Thanks for the tip, I've made this change. 

It did cause me problems in one case. There are times when I want to do:
```
assert.equal(syncResult.status, 'complete');
assert(!syncResult?.hadJsonErrorsOrWarnings);
```

But this gives a type error because of our new definitions of the `SyncResult` type (even when using option chaining!). 

So instead I ended up having to do:

```
assert.equal(syncResult.status, 'complete');
assert(syncResult.status === 'complete' && !syncResult.hadJsonErrorsOrWarnings);
```
so that I will get all the needed info and the Typechecker is happy. Let me know if there is something better I could do.

Reply:
Ahh, that's a good point. Given that, I think I'd rather do what you were doing originally. As a middle ground, you could also split into two assertions, which I think TypeScript would be happy with:

```
assert(syncResult.status === 'complete');
assert.isFalse(syncResult.hadJsonErrorsOrWarnings);
```

Reply:
Unfortunately, that still gives a type error. It seems that the typechecker can't infer the type narrowing of `syncResult` based on the `assert` the way that it would if it were an `if` statement.